### PR TITLE
fix 15367 - can't edit admin-user

### PR DESCRIPTION
**Pull request by KRav007**
Fixed issue # : 15367: 
New feature # :
Changed feature # :
Dev: 
Dev:

### CODE REVIEW

File: application/controllers/admin/UserManagement.php [hunk 1]
```diff
@@ -873,30 +882,19 @@ private function _createNewUser($aUser, $sendMail = true)
...
     /**
-     * private method to store a change in a user
-     * return json to return to js frontend
+     * Update admin-user
      *
-     * @param array $aUser
-     * @param integer $userid
-     * @return string
+     * @param $aUser
+     * @return array|bool|mixed|null
```
Comment (Shnoulle) [inline application/controllers/admin/UserManagement.php]: `return false|\User`

File: application/controllers/admin/UserManagement.php [hunk 2]
```diff
@@ -873,30 +882,19 @@ private function _createNewUser($aUser, $sendMail = true)
...
-
         $oUser->setAttributes($aUser);
         $oUser->modified = date('Y-m-d H:i:s');
 
-        $aUser = [];
         if ($oUser->save()) {
-            return $oUser->attributes();
+            return $oUser;
         }
         return false;
```
Comment (Shnoulle) [inline application/controllers/admin/UserManagement.php]: No you must always return Model to get the errors

File: application/controllers/admin/UserManagement.php [hunk 3]
```diff
@@ -169,17 +170,17 @@ public function applyedit()
             Yii::app()->end();
         }
 
-        $aUser = $this->_editUser($aUser);
-        if ($aUser === false) {
+        $oUser = $this->updateAdminUser($aUser);
+        if ($oUser === false) {
```
Comment (Shnoulle) [inline application/controllers/admin/UserManagement.php]: if $oUser->hasErrors() https://www.yiiframework.com/doc/api/1.1/CModel#hasErrors-detail

File: application/controllers/admin/UserManagement.php [hunk 4]
```diff
@@ -169,17 +170,17 @@ public function applyedit()
...
             return Yii::app()->getController()->renderPartial('/admin/usermanagement/partial/json', ["data" => [
                 'success' => false,
                 'error' => print_r($oUser->getErrors(), true),
```
Comment (Shnoulle) [inline application/controllers/admin/UserManagement.php]: Here $oUser === false , it's NOT a object

File: application/controllers/admin/UserManagement.php (hunk 2 above)
Comment (KRav007) [reply application/controllers/admin/UserManagement.php]: ok

File: application/controllers/admin/UserManagement.php (hunk 3 above)
Comment (KRav007) [reply application/controllers/admin/UserManagement.php]: ok

File: application/controllers/admin/UserManagement.php (hunk 4 above)
Comment (KRav007) [reply application/controllers/admin/UserManagement.php]: ok

File: application/controllers/admin/UserManagement.php (hunk 1 above)
Comment (KRav007) [reply application/controllers/admin/UserManagement.php]: ok

File: application/controllers/admin/UserManagement.php [hunk 5]
```diff
@@ -880,23 +883,20 @@ private function _createNewUser($aUser, $sendMail = true)
...
     {
         $oUser = User::model()->findByPk($aUser['uid']);
         $oUser->setAttributes($aUser);
         $oUser->modified = date('Y-m-d H:i:s');
 
-        if ($oUser->save()) {
-            return $oUser;
-        }
-        return false;
+        return  $oUser->save();
```
Comment (Shnoulle) [inline application/controllers/admin/UserManagement.php]: It must not be 
````
$oUser->save();
return $oUser;
```` 

?

**Pull request merged**